infra.bs

```diff
@@ -872,6 +886,156 @@ For <a>pairs</a>, a slightly shorter literal syntax can be used, separating the
 as 200/`<code>OK</code>`.
 
 
+<h2 id=forgiving-base64>Forgiving base64</h2>
+
+<p>To <dfn export>forgiving-base64 encode</dfn> given a <a>byte sequence</a> <var>data</var>, apply
+the base64 algorithm defined in section 4 of RFC 4648 to <var>data</var> and return the result.
+[[!RFC4648]]
+
+<p class="note no-backref">This is named <a>forgiving-base64 encode</a> for symmetry with
+<a>forgiving-base64 decode</a>, which is different from the RFC as it defines error handling for
+certain inputs.
+
+<p>To <dfn export>forgiving-base64 decode</dfn> given a string <var>data</var>, run these steps:</p>
```

Comment:
It defines an Encoding scheme and some rules around it. The idea is that you infer the decode and encode algorithms from that. Similar to using ABNF and expecting you have a parser that works.